test/ng/compileSpec.js

```diff
@@ -3373,6 +3373,90 @@ describe('$compile', function() {
     });
   });
 
+  describe('collector', function() {
+
+    var collected;
+    beforeEach(module(function($compileProvider) {
+      collected = false;
+      $compileProvider.directive('testCollect', function() {
+        return {
+          restrict: 'EACM',
+          link: function() {
+            collected = true;
+          }
+        };
+      });
+    }));
+
+    it('should collect comment directives by default', inject(function() {
+      var html = '<!-- directive: test-collect -->';
+      element = $compile('<div>' + html + '</div>')($rootScope);
+      expect(collected).toBe(true);
+    }));
+
+    it('should collect css class directives by default', inject(function() {
+      element = $compile('<div class="test-collect"></div>')($rootScope);
+      expect(collected).toBe(true);
+    }));
+
+    angular.forEach([
+      { commentEnabled: true, cssEnabled: true },
+      { commentEnabled: true, cssEnabled:false },
+      { commentEnabled:false, cssEnabled: true },
+      { commentEnabled:false, cssEnabled:false }
```

Comment:
Nitpick: Inconsistent spacing. (Always have a space after `:`, preferrable to have no space after `{` or before `}`.)